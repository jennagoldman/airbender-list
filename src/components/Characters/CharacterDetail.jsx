import React from 'react';
import PropTypes from 'prop-types';
import styles from './CharacterDetail.css';

const CharacterDetail = ({ name, affiliation, allies, enemies, photoUrl }) => (
  <div className={styles.characterDetail}>
    <p className={styles.characterName}>{name}</p>
    <img src={photoUrl} alt={name} />
    <p><i className="fas fa-users"></i> {affiliation} </p>
    <p>
      <span><i className="fas fa-hands-helping"></i> {allies} | </span>
      <span><i className="fas fa-fist-raised"></i> {enemies}</span>
    </p>
  </div>
);

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string,
  allies: PropTypes.array,
  enemies: PropTypes.array,
  affiliation: PropTypes.string,
};


export default CharacterDetail;
