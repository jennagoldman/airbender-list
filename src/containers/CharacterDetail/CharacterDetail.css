.characterDetail {
  background: #fff;
  width: 400px;
  height: 450px;
  padding: 25px 20px;
  border-radius: 15px;
  box-shadow: 10px 10px #000;
  margin: 0 auto;
}

.characterDetail img {
  width: 80%;
  max-height: 250px;
}

.characterName {
  color: #ff7050;
  font-family: 'Lemonada', sans-serif;
  text-shadow: 1px 1px #000;
}

i {
  color: #ff7050
}