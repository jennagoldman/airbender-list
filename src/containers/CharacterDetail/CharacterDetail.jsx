import React, { useState, useEffect } from 'react';
import { fetchCharacter } from '../../services/airbender-api.js';
import { useRouteMatch } from 'react-router-dom';
import styles from './CharacterDetail.css';

const CharacterDetail = () => {
  const [character, setCharacter] = useState({});

  const match = useRouteMatch('/character/:id');

  useEffect(() => {
    fetchCharacter(match.params.id)
      .then(character => setCharacter(character));
  }, []);

  return (
    <div className={styles.characterDetail}>
      <p className={styles.characterName}>{character.name}</p>
      <img src={character.photoUrl} alt={character.name} />
      <p><i className="fas fa-users"></i> {character.affiliation} </p>
      <p>
        <span><i className="fas fa-hands-helping"></i> {character.allies} | </span>
        <span><i className="fas fa-fist-raised"></i> {character.enemies}</span>
      </p>
    </div>
  );
};

export default CharacterDetail;
