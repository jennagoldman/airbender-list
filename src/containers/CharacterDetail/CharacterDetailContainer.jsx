import React, { useState, useEffect } from 'react';
import { fetchCharacter } from '../../services/airbender-api.js';
import { useRouteMatch } from 'react-router-dom';
import CharacterDetail from '../../components/Characters/CharacterDetail.jsx';

const CharacterDetailContainer = () => {
  const [character, setCharacter] = useState({});

  const match = useRouteMatch('/character/:id');

  useEffect(() => {
    fetchCharacter(match.params.id)
      .then(character => setCharacter(character));
  }, []);

  return (
    <>
      <CharacterDetail {...character} />
    </>
  );
};

export default CharacterDetailContainer;
