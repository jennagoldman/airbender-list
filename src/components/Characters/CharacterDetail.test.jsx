import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetail from './CharacterDetail.jsx';

describe('CharacterDetail', () => {
  it('matches a snapshot', () => {

    const wrapper = shallow(<CharacterDetail name="Momo" photoUrl="https://vignette.wikia.nocookie.net/avatar/images/4/43/Inquisitive_Momo.png/revision/latest?cb=20081225191217" affliation="Team Aang" allies={['Aang']} enemies={['Zuko']} />);
    expect(wrapper).toMatchSnapshot();
  });
});
