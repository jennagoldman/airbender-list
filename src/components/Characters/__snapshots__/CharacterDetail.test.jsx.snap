// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail matches a snapshot 1`] = `
<div
  className="characterDetail"
>
  <p
    className="characterName"
  >
    Momo
  </p>
  <img
    alt="Momo"
    src="https://vignette.wikia.nocookie.net/avatar/images/4/43/Inquisitive_Momo.png/revision/latest?cb=20081225191217"
  />
  <p>
    <i
      className="fas fa-users"
    />
     
     
  </p>
  <p>
    <span>
      <i
        className="fas fa-hands-helping"
      />
       
      Aang
       | 
    </span>
    <span>
      <i
        className="fas fa-fist-raised"
      />
       
      Zuko
    </span>
  </p>
</div>
`;
